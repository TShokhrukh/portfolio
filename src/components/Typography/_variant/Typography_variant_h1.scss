.Typography_variant_h1 {
  --font-size: #{$font-size-h1};
}
