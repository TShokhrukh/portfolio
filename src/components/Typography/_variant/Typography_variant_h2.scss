.Typography_variant_h2 {
  --font-size: #{$font-size-h2};
}
