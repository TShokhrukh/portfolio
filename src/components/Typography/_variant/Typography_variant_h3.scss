.Typography_variant_h3 {
  --font-size: #{$font-size-h3};
}
