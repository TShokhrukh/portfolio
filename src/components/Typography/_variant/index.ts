export * from './Typography_variant_body'
export * from './Typography_variant_span'
export * from './Typography_variant_h1'
export * from './Typography_variant_h2'
export * from './Typography_variant_h3'
export * from './Typography_variant_h4'
export * from './Typography_variant_h5'
export * from './Typography_variant_h6'
