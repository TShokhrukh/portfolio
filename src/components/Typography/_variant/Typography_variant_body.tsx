import React from 'react'
import { withBemMod } from '@bem-react/core'
import { cnTypography } from '../Typography.const'
import type { ITypographyProps, ITypographyVariantBodyProps } from '../__types__'

import './Typography_variant_body.scss'

export const withTypographyVariantBody = withBemMod<ITypographyVariantBodyProps, ITypographyProps>(
  cnTypography(),
  { variant: 'body' },
  (Typography) => function TypographyVariantBody (props) {
    return <Typography {...props} component='p' variant={undefined} />
  }
)
