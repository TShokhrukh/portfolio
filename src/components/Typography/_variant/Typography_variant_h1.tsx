import React from 'react'
import { withBemMod } from '@bem-react/core'
import { cnTypography } from '../Typography.const'
import type { ITypographyProps, ITypographyVariantH1Props } from '../__types__'

import './Typography_variant_h1.scss'

export const withTypographyVariantH1 = withBemMod<ITypographyVariantH1Props, ITypographyProps>(
  cnTypography(),
  { variant: 'h1' },
  (Typography) => function TypographyVariantH1 (props) {
    return <Typography {...props} component='h1' variant={undefined} />
  }
)
