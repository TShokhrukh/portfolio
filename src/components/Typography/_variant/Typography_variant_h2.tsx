import React from 'react'
import { withBemMod } from '@bem-react/core'
import { cnTypography } from '../Typography.const'
import type { ITypographyProps, ITypographyVariantH2Props } from '../__types__'

import './Typography_variant_h2.scss'

export const withTypographyVariantH2 = withBemMod<ITypographyVariantH2Props, ITypographyProps>(
  cnTypography(),
  { variant: 'h2' },
  (Typography) => function TypographyVariantH2 (props) {
    return <Typography {...props} component='h2' variant={undefined} />
  }
)
