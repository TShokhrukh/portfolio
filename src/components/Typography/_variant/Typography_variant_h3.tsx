import React from 'react'
import { withBemMod } from '@bem-react/core'
import { cnTypography } from '../Typography.const'
import type { ITypographyProps, ITypographyVariantH3Props } from '../__types__'

import './Typography_variant_h3.scss'

export const withTypographyVariantH3 = withBemMod<ITypographyVariantH3Props, ITypographyProps>(
  cnTypography(),
  { variant: 'h3' },
  (Typography) => function TypographyVariantH3 (props) {
    return <Typography {...props} component='h3' variant={undefined} />
  }
)
