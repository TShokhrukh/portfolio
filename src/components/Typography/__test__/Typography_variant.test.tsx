/* global test, expect, describe */
import React from 'react'
import { render, screen } from '@testing-library/react'
import { compose, composeU } from '@bem-react/core'
import { Typography as TypographyBase } from '../Typography'
import { withTypographyVariantBody } from '../_variant/Typography_variant_body'
import { withTypographyVariantSpan } from '../_variant/Typography_variant_span'
import { withTypographyVariantH1 } from '../_variant/Typography_variant_h1'
import { withTypographyVariantH2 } from '../_variant/Typography_variant_h2'
import { withTypographyVariantH3 } from '../_variant/Typography_variant_h3'
import { withTypographyVariantH4 } from '../_variant/Typography_variant_h4'
import { withTypographyVariantH5 } from '../_variant/Typography_variant_h5'
import { withTypographyVariantH6 } from '../_variant/Typography_variant_h6'

export const Typography = compose(
  composeU(
    withTypographyVariantBody,
    withTypographyVariantSpan,
    withTypographyVariantH1,
    withTypographyVariantH2,
    withTypographyVariantH3,
    withTypographyVariantH4,
    withTypographyVariantH5,
    withTypographyVariantH6
  )
)(TypographyBase)

const variants = [
  'body',
  'span',
  'h1',
  'h2',
  'h3',
  'h4',
  'h5',
  'h6'
] as const

describe.each(variants)('Typography variant %s', (variant) => {
  test(`must return the component template for variant ${variant}`, () => {
    render(<Typography variant={variant} data-testid='typography'>children</Typography>)

    expect(screen.getByTestId('typography')).toMatchSnapshot()
  })
})
