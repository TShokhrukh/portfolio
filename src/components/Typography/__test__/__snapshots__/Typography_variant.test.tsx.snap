// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography variant body must return the component template for variant body 1`] = `
<p
  class="Typography Typography_variant_body"
  data-testid="typography"
>
  children
</p>
`;

exports[`Typography variant h1 must return the component template for variant h1 1`] = `
<h1
  class="Typography Typography_variant_h1"
  data-testid="typography"
>
  children
</h1>
`;

exports[`Typography variant h2 must return the component template for variant h2 1`] = `
<h2
  class="Typography Typography_variant_h2"
  data-testid="typography"
>
  children
</h2>
`;

exports[`Typography variant h3 must return the component template for variant h3 1`] = `
<h3
  class="Typography Typography_variant_h3"
  data-testid="typography"
>
  children
</h3>
`;

exports[`Typography variant h4 must return the component template for variant h4 1`] = `
<h4
  class="Typography Typography_variant_h4"
  data-testid="typography"
>
  children
</h4>
`;

exports[`Typography variant h5 must return the component template for variant h5 1`] = `
<h5
  class="Typography Typography_variant_h5"
  data-testid="typography"
>
  children
</h5>
`;

exports[`Typography variant h6 must return the component template for variant h6 1`] = `
<h6
  class="Typography Typography_variant_h6"
  data-testid="typography"
>
  children
</h6>
`;

exports[`Typography variant span must return the component template for variant span 1`] = `
<span
  class="Typography Typography_variant_span"
  data-testid="typography"
>
  children
</span>
`;
